cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/PullStrategyTest.java

```diff
@@ -76,6 +78,56 @@ private void pull(PullStrategy replicator, int expectedDocs) throws Exception {
     }
 
     @Test
+    public void pull_changesNewerThanOpenRevs() throws Exception {
+        // test for regressions against the following scenario:
+        // - changes feed tells us that 1-rev is a new rev we don't have
+        // - on remote db, 2-rev gets added and becomes the new leaf
+        // - compact() is called and 1-rev is compacted away
+        // - we try to pull 1-rev but it is no longer there
+        // the solution to this is to insert a stub revision into the tree
+        // and continue as normal.
+
+        // upload some docs
+        Bar bar1 = BarUtils.createBar(remoteDb, "Tom", 31);
+
+        // capture 'real' changes feed - after first rev
+        final ChangesResult changesResult1 = remoteDb.changes(null, 1000);
+
+        // upload some more docs
+        Bar bar2 = BarUtils.updateBar(remoteDb, bar1.getId(), "Jerry", 41);
+
+        // capture 'real' changes feed - after second rev
+        final ChangesResult changesResult2 = remoteDb.changes(null, 1000);
+
+        // set up replication...
+        PullStrategy customReplicator = super.getPullStrategy();
+
+        // inject our mocked changesresult via a 'spy'
+        // ChangesResultAnswer returns a missing revision each time.
+        customReplicator.sourceDb = spy(customReplicator.sourceDb);
+        doAnswer(new ChangesResultAnswer(changesResult1, changesResult2))
+                .when(customReplicator.sourceDb).
+                changes((PullFilter)
+                        anyObject(), anyObject(), anyInt());
+        // do the actual replication
+        customReplicator.run();
+
```

Comment:
So this test is useful because it checks that the replicator doesn't explode with an exception. But we should have a bunch of asserts at the end to make sure we got all of the expected documents, that the sequence number is correct, etc.